crates/metadata-store/src/rocksdb/tests.rs

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2024 - Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::rocksdb::client::RocksDBMetadataStoreClient;
+use crate::rocksdb::server::RocksDBMetadataStore;
+use crate::{MetadataStoreClient, WriteError};
+use bytestring::ByteString;
+use futures::stream::FuturesUnordered;
+use futures::StreamExt;
+use restate_core::{MockNetworkSender, TaskKind, TestCoreEnv, TestCoreEnvBuilder};
+use restate_types::metadata_store::Precondition;
+use restate_types::{Version, Versioned};
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+use test_log::test;
+
+#[derive(Debug, Clone, PartialOrd, PartialEq, Serialize, Deserialize, Default)]
+struct Value {
+    version: Version,
+    value: String,
+}
+
+impl Value {
+    fn next_version(mut self) -> Self {
+        self.version = self.version.next();
+        self
+    }
+}
+
+impl Versioned for Value {
+    fn version(&self) -> Version {
+        self.version
+    }
+}
+
+/// Tests basic operations of the metadata store.
+#[test(tokio::test)]
+async fn basic_metadata_store_operations() -> anyhow::Result<()> {
+    let (client, env) = create_test_environment().await?;
+
+    env.tc
+        .run_in_scope("test", None, async move {
```

Comment:
One day we will have cycles to convert this into a macro that replaces (`tokio::test`) 😄

Reply:
Ui, that would be cool. Would also be a great learning opportunity for me because I am a total macro noob.